public/app/features/variables/state/actions.ts

```diff
@@ -555,10 +562,12 @@ export const initVariablesTransaction = (dashboardUid: string, dashboard: Dashbo
     dispatch(variablesInitTransaction({ uid: dashboardUid }));
 
     dispatch(initDashboardTemplating(dashboard.templating.list));
-    await dispatch(processVariables());
+
     dispatch(completeDashboardTemplating(dashboard));
 
     dispatch(variablesCompleteTransaction({ uid: dashboardUid }));
+
+    await dispatch(processVariables());
```

Comment:
this needs to where it is, as the two actions above rely on variable queries having been executed

Reply:
`processVariables` needs to be dispatched after `completeDashboardTemplating` . than only it's refreshing the variable on dashboard load 
shall we call the function after completeDashboardTemplating  ?